### ISSUE TITLE
[Feature request] Override vimtex's main file detection upon the presence of @default_files or @default_excluded_files in project-specific latexmkrc

### ISSUE THREAD

**Issue by krishnakumarg1984** [lervag/vimtex#1325]
If vimtex is set to use `latexmk`, the project-specific `latexmkrc` takes precedence (if present) over any conflicting vimtex configuration. The same principle should hold true for main-file detection.

In particular two variables `@default_files` and `@default_excluded_files` can be present in a `latexmkrc` file that governs which files must be compiled.  For example, with the line

>@default_files = ('main.tex')

it is sufficient to just type `latexmk` at the command prompt without any additional arguments, which is quite convenient.  In a big project like a PhD thesis spread over 30-40 files, it is inconvenient to correctly provide the relative path `%! TEX root  = ../../main.tex` or `%! TEX root  = ../main.tex` or `%! TEX root  = ../../../main.tex`  in __each__ file suitably. 

Speaking from personal experience, imagine that during the course of the thesis writing, some of `tex` files moved in the filesystem tree from `chapters/introduction/images/` to `chapters/images` or from `chapters/peripheral content/backmatter` to just `backmatter`, we have to figure out the right path for each `tex` file in the relevant sub-tree suitably. Same goes for renaming. If we rename the main file in the course of 6 months of thesis writing from `main.tex` to `thesis.tex` and then again to `project.tex`, and finally back to `main.tex`, imagine renaming the buffer variable at the top of each relevant file (I know it's not that difficult with `cdo`, but nevertheless unnecessary).

Clearly, the practice of having a single place (in `latexmkrc`) to define the `main.tex` file is superior to declaring it in 48 different files and having all of them to be in sync). 

The excluded files is an interesting case. Configuring, for example
>@default_files = ('not_this.tex','not_this_either.tex')

will cause a simple `latexmk` invocation to compile every other `tex` file in the current directory, other than the above named files.  This is particularly helpful when the user would like to compile two different versions of a document.  An interesting use case is for the CV/Resume wherein two different page layouts are needed (for example A4 paper & letter paper), with corresponding little tweaks in their source codes.
eg. https://github.com/krishnakumarg1984/resume_CV , but perhaps other `tex` files are present in the root directory which are inflexible to be moved elsewhere to a sub-directory. 

Now, simply running `latexmk` in that repository's root shall produce the A4 and Letter versions of the CV.

In summary, please just parse latexmkrc for `@default_files`/`@default_excluded_files`  & override `vimtex's` main file detection.

**Comment by lervag** [comment]
Ok, I understand your request. My initial thought is that I disagree. But it would help very much if you could create a good, minimal example where such a feature would be helpful.

There are several reasons why I disagree:
- The "general" method of finding the main file should already work for most cases. I think it should be easier to adapt your project to this method, then to adapt vimtex to every possible edge case.
- With this method, we need to account for a lot of edge cases, such as i) we should only look for the `@default_files` option in a local `.latexmkrc` file, not in any global files, and ii) if the `@default_files` is found, but the detected file is not readable, how should we proceed?
- The `@default_files` option is there to specify which files to compile. When you open a file with vimtex, it should belong to a single project, and we should not compile more than a single project. That is, for your CV example, we should never compile two documents from a single document. If you want to do this, then I advice you to do it with other tools. Adding such support is way too complicated for me to consider.

Note: If you want to work with the `@default_files` option as described in your CV example, I would propose that you use something like [`entr`](http://eradman.com/entrproject/), e.g.:

1. Edit your latex project in a Vim instance (or whatever).
2. In a different terminal, run: `ls *|entr -c latexmk`.

**Comment by krishnakumarg1984** [comment]
@lervag ok. Please feel free to close this issue. A

**Comment by lervag** [comment]
Thanks!
